crates/turbopack-ecmascript/src/tree_shake/chunk_item.rs

```diff
@@ -27,16 +28,32 @@ pub struct EcmascriptModulePartChunkItem {
 #[turbo_tasks::value_impl]
 impl EcmascriptChunkItem for EcmascriptModulePartChunkItem {
     #[turbo_tasks::function]
-    async fn content(&self) -> Result<EcmascriptChunkItemContentVc> {
-        let module = self.module.await?;
+    fn content(self_vc: EcmascriptModulePartChunkItemVc) -> EcmascriptChunkItemContentVc {
+        self_vc.content_with_availability_info(Value::new(AvailabilityInfo::Untracked))
+    }
+
+    #[turbo_tasks::function]
+    async fn content_with_availability_info(
+        self_vc: EcmascriptModulePartChunkItemVc,
+        availability_info: Value<AvailabilityInfo>,
+    ) -> Result<EcmascriptChunkItemContentVc> {
+        let this = self_vc.await?;
+        let availability_info = if *this.module.analyze().need_availability_info().await? {
```

Comment:
Is there harm in just always passing down the availability info?

Reply:
cache key of turbo functions are the arguments. So if it would pass down the availability info even when not needed that would duplicate the work. If not passing down availability info it can share the cache. This wrapper function will make sure we can leverage more caching when availability info isn't needed.